File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--add-namedrange-controls-to-worksheets_2.cs
        private void AddNamedRange()
        {
            Microsoft.Office.Tools.Excel.NamedRange textInCell;

            Worksheet worksheet = Globals.Factory.GetVstoObject(
                Globals.ThisAddIn.Application.ActiveWorkbook.Worksheets[1]);


            Excel.Range cell = worksheet.Range["A1"]; 
            textInCell = worksheet.Controls.AddNamedRange(cell, "MyNamedRange");
            textInCell.Value2 = "Hello World";
         }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--add-namedrange-controls-to-worksheets_2.vb
    Private Sub AddNamedRange()
        Dim textInCell As Microsoft.Office.Tools.Excel.NamedRange

        Dim NativeWorksheet As Microsoft.Office.Interop.Excel.Worksheet =
            Globals.ThisAddIn.Application.ActiveWorkbook.Worksheets(1)

        Dim worksheet As Microsoft.Office.Tools.Excel.Worksheet =
            Globals.Factory.GetVstoObject(NativeWorksheet)

        Dim cell As Excel.Range = worksheet.Range("A1")
        textInCell = worksheet.Controls.AddNamedRange(cell, "MyNamedRange")
        textInCell.Value2 = "Hello World"
    End Sub